libbeat/autodiscover/builder/helper.go

```diff
@@ -155,14 +155,14 @@ func IsNoOp(hints common.MapStr, key string) bool {
 }
 
 // GenerateHints parses annotations based on a prefix and sets up hints that can be picked up by individual Beats.
-func GenerateHints(annotations common.MapStr, container, prefix string) common.MapStr {
+func GenerateHints(annotations common.MapStr, container, prefix string, separator string) common.MapStr {
 	hints := common.MapStr{}
 	if rawEntries, err := annotations.GetValue(prefix); err == nil {
 		if entries, ok := rawEntries.(common.MapStr); ok {
 			for key, rawValue := range entries {
 				// If there are top level hints like co.elastic.logs/ then just add the values after the /
 				// Only consider namespaced annotations
-				parts := strings.Split(key, "/")
+				parts := strings.Split(key, separator)
```

Comment:
Would this break if the chosen separator is `.`? we may need to add a check for the accepted separators. This may need some more tests too.